### PR TITLE
Discord Integration

### PR DESCRIPTION
**Pull request by roddajohn**
This is totally in progress.

Things todo as of now:
- [x] Add a Discord documentation page
- [x] Add more useful messages in Discord with regard to the commands available

### CODE REVIEW

**Comment by roddajohn** [comment]
I think this is ready:

Working on the deployer now.

**Comment by roddajohn** [comment]
Actually, let's hold off for at least a few hours as I figure out the docker-compose stuff related to the deploy.

File: Dockerfile.discord~ [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM python:3.7
```
Comment (BenMusch) [inline Dockerfile.discord~:1]: What's the difference between this file and `Dockerfile.discord`? Do we need both?

File: mit_tab_bot/api.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    return api
+
+
+def get_judge(discord_id):
+    api = construct_api()
+
+    response = api.judges.list(
+        params={'discord_id': discord_id}
+    )
+    return response.body[0] if len(response.body) > 0 else None
```
Comment (BenMusch) [inline mit_tab_bot/api.py:27]: nit: can just be `if response.body` or `if len(response.body)`

File: mit_tab_bot/roles.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+from .api import (
+    is_judge,
+    is_debater,
+    get_judge,
+    get_debater
+)
+
+ADMIN = [
+    'Talos#5956'
```
Comment (BenMusch) [inline mit_tab_bot/roles.py:11]: Can you explain what exactly this corresponds to? Is this your personal user id or something?

**Review by BenMusch — COMMENTED** [review]
High-level comment: What do you think about shipping `mit_tab_bot/*` as a separate repo? It can build and public to dockerhub and then we would reference that in `docker-compose`

**Comment by roddajohn** [comment]
@BenMusch on the high level: yup that makes sense I'll start working on this, everything else is inline

File: mit_tab_bot/api.py (hunk 2 above)
Comment (roddajohn) [reply mit_tab_bot/api.py:27]: lol yes

File: mit_tab_bot/roles.py (hunk 3 above)
Comment (roddajohn) [reply mit_tab_bot/roles.py:11]: Yeah this is my personal Discord ID which grants superuser, plan is to make this part of the form on the deployer, and passed through an ENV variable, but I haven't finished the deployer piece yet so here it is.

Comment (BenMusch) [reply mit_tab_bot/roles.py:11]: Gotcha, yeah would be nice to have this set via some env var/config file and then referenced by a constant like `ADMIN_DISCORD_ID` to make things clear

**Pull request closed without merging**